doc/RECOVER.md

```diff
@@ -0,0 +1,162 @@
+
+Here we need to process 2 descriptors from the original one, replacing `<0;1>` by `0` for the receive descriptor and `1` for change descriptor, checksum might to be updated for each ones.
+We also need to replace `'` used for defined hardened derivation path by `h`.
+
+#### Processed descriptors:
+
+wsh(or_d(pk([a5c6b76e/48`h`/1`h`/0`h`/2`h`]tpubDF5861hj6vR3iJr3aPjGJz4rNbqDCRujQ21mczzKT5SiedaQqNVgHC8HT9ceyxvMFRoPMx4P6HAcL3NZrUPhRUbwCyj3TKSa64bAfnE3sLh/`0`/\*),and_v(v:pkh([c477fd13/48`h`/1`h`/0`h`/2`h`]tpubDFn7iPbFqGrTQ2aRACNsUK1MXQR4Z6dYfU2nD1WA9ifSaia642j3Wah4n5pBUEpERNWGJsyv3Dv5qwBabC9TLQrwSboKzukw9wmurGu7XVH/`0`/\*),older(3))))
+
+wsh(or_d(pk([a5c6b76e/48`h`/1`h`/0`h`/2`h`]tpubDF5861hj6vR3iJr3aPjGJz4rNbqDCRujQ21mczzKT5SiedaQqNVgHC8HT9ceyxvMFRoPMx4P6HAcL3NZrUPhRUbwCyj3TKSa64bAfnE3sLh/`1`/\*),and_v(v:pkh([c477fd13/48`h`/1`h`/0`h`/2`h`]tpubDFn7iPbFqGrTQ2aRACNsUK1MXQR4Z6dYfU2nD1WA9ifSaia642j3Wah4n5pBUEpERNWGJsyv3Dv5qwBabC9TLQrwSboKzukw9wmurGu7XVH/`1`/\*),older(3))))
+
+Now we got the descriptors without checksums, the command to pass to bitcoin-cli should look like this:
+
+bitcoin-cli -signet -rpcwallet=liana importdescriptors '[{"desc": "`<descriptor_receive>`", "range": [0,10000], "timestamp": 1682920310}, {"desc": "`<descriptor_change>`", "range": [0,10000], "timestamp": 1682920310}]'
+
+where we replace `<descriptor_receive>` and `<descriptor_change>` by relevant descriptors (with old/wrong checksum):
+
+```shell
+bitcoin-cli -signet -rpcwallet=liana importdescriptors '[{"desc": "wsh(or_d(pk([a5c6b76e/48h/1h/0h/2h]tpubDF5861hj6vR3iJr3aPjGJz4rNbqDCRujQ21mczzKT5SiedaQqNVgHC8HT9ceyxvMFRoPMx4P6HAcL3NZrUPhRUbwCyj3TKSa64bAfnE3sLh/0/*),and_v(v:pkh([c477fd13/48h/1h/0h/2h]tpubDFn7iPbFqGrTQ2aRACNsUK1MXQR4Z6dYfU2nD1WA9ifSaia642j3Wah4n5pBUEpERNWGJsyv3Dv5qwBabC9TLQrwSboKzukw9wmurGu7XVH/0/*),older(3))))#8ldsjayd", "range": [0,10000], "timestamp": 1682920310, "active": true, "internal":false}, {"desc": "wsh(or_d(pk([a5c6b76e/48h/1h/0h/2h]tpubDF5861hj6vR3iJr3aPjGJz4rNbqDCRujQ21mczzKT5SiedaQqNVgHC8HT9ceyxvMFRoPMx4P6HAcL3NZrUPhRUbwCyj3TKSa64bAfnE3sLh/1/*),and_v(v:pkh([c477fd13/48h/1h/0h/2h]tpubDFn7iPbFqGrTQ2aRACNsUK1MXQR4Z6dYfU2nD1WA9ifSaia642j3Wah4n5pBUEpERNWGJsyv3Dv5qwBabC9TLQrwSboKzukw9wmurGu7XVH/1/*),older(3))))#8ldsjayd", "range": [0,10000], "timestamp": 1682920310, "active": true, "internal":true}]'
+```
+
+output:
+``` 
+[
+  {
+    "success": false,
+    "error": {
+      "code": -5,
+      "message": "Provided checksum '8ldsjayd' does not match computed checksum 'rgmh0m0p'"
+    }
+  },
+  {
+    "success": false,
+    "error": {
+      "code": -5,
+      "message": "Provided checksum '8ldsjayd' does not match computed checksum 'u7dlz0cc'"
```

Comment:
It would be much cleaner to call `getdescriptorinfo` and get the checksum from there. Triggering an error on purpose is confusing and error messages are usually unreliable.

Reply:
right, i forgot to fix that, prior to finding out there's a bug in importdescriptor (`'` are dropped and not count as hardened path) the checksum  from getdescriptorinfo was not matching, so i decide to go this ugly way instead